packages/client/src/schema/pagination.ts

```diff
@@ -104,11 +104,50 @@ export function isCursorPaginationOptions(
 }
 
 export class RecordArray<Result extends XataRecord> extends Array<Result> {
+  constructor(overrideRecords?: Result[]);
+  constructor(...args: any[]) {
+    super(...RecordArray.parseConstructorParams(...args));
+  }
+
+  static parseConstructorParams(...args: any[]) {
+    // new <T>(arrayLength: number): T[]
+    if (args.length === 1 && typeof args[0] === 'number') {
+      return new Array(args[0]);
+    }
+
+    // new <T>(overrideRecords: Array | undefined): T[>]
+    if (args.length <= 1 && Array.isArray(args[0] ?? [])) {
+      const result = args[0] ?? [];
+      return new Array(...result);
+    }
+
+    // <T>(...items: T[]): T[]
+    return new Array(...args);
+  }
+
+  toArray(): Result[] {
+    return new Array(...this);
+  }
+
+  toSerializable(): JSONData<Result>[] {
+    return JSON.parse(this.toString());
+  }
+
+  toString(): string {
+    return JSON.stringify(this.toArray());
+  }
+
+  map<U>(callbackfn: (value: Result, index: number, array: Result[]) => U, thisArg?: any): U[] {
```

Comment:
What is `thisArg`?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#thisarg

Reply:
Ah thanks. Never used that argument